### ISSUE TITLE
TouchBarButton: change touchBarButton.icon not working in 2.0.0

### ISSUE THREAD

**Issue by g1eny0ung** [electron/electron#12872]
* Electron Version: 2.0.0
* Operating System (Platform and Version): MacOS 10.13.4
* Last known working Electron version: 1.8.4

**Expected Behavior**

This is my application's part of code about changing **touchBarButton.icon**

```
ipcMain.on('touchBarPauseAndStart', (event, arg) => {
  if (arg === true) {
    tb.pauseAndStart.icon = __dirname + '/resources/pause.png'
    tb.touchBarState.pause = false
  } else {
    tb.pauseAndStart.icon = __dirname + '/resources/play.png'
    tb.touchBarState.pause = true
  }
})
```

I expect my currently playing audio to pause or play, and also through ipc to change the touchbarbutton's icon.

**Actual behavior**

This behavior worked fine in electron version 1.8.4, but when I upgrade electron to 2.0.0, seems that it doesn't working as before.

**To Reproduce**

This is the source code repo. [https://github.com/ilime/Petal](https://github.com/ilime/Petal)

Now I downgrade the electron version to 1.8.4.

Code refer to this issue:

[ipc.js: Line 5](https://github.com/ilime/Petal/blob/74623929adc0d7cfc99ec9df925d74d2c233be22/electron/ipc.js#L5)

[Cover/index.jsx: Line 40](https://github.com/ilime/Petal/blob/74623929adc0d7cfc99ec9df925d74d2c233be22/src/components/FM/Cover/index.jsx#L40)

[Cover/index.jsx: Line 148](https://github.com/ilime/Petal/blob/74623929adc0d7cfc99ec9df925d74d2c233be22/src/components/FM/Cover/index.jsx#L148)

**Comment by ckerr** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

Would it be possible for you to fork [electron-quick-start](https://github.com/electron/electron-quick-start) to make a  small test case to reproduce the issue in isolation?

Standalone test cases makes fixing issues go more smoothly: not only does it ensure testers and reporters are looking at the same issue, but it also prevents testers from the slower task of triaging an unfamiliar codebase.

Thanks in advance! Your help is appreciated.

**Comment by g1eny0ung** [comment]
@ckerr Sure, it's my fault. I will try to reproduce this problem in the quick-start repo later.

**Comment by g1eny0ung** [comment]
I found what causes this issue.

This is the test repo. [https://github.com/g1eny0ung/electron-quick-start/tree/touchBarButton](https://github.com/g1eny0ung/electron-quick-start/tree/touchBarButton)

First, I send a message from renderer.js.

```
const { ipcRenderer } = require('electron')

let status = true

ipcRenderer.on('pause', send)

function send() {
  ipcRenderer.send('touchBarPauseAndStart', status)
  status = !status
}

document.addEventListener('click', send)
```

In [main.js: Line 38](https://github.com/g1eny0ung/electron-quick-start/blob/fe41918761ffd348c76f4b56dc5b608ccb5d2452/main.js#L38), when I set touchBar only once, all works fine in version 2.0.0 and 1.8.4. The touchBarButton.icon was succeeded switching.

But when I set touchBar twice, only 1.8.4 worked, seems like in version 2.0.0 the last touchBar covers up the previous touchBar? I'm not very sure.

I found this because I repeatedly set touchBar in my app. In my app, it's wrong to set twice. But I'm not sure this is or not a **bug** in electron?

Thanks for reply ❤️ .
